lib/matplotlib/figure.py

```diff
@@ -949,13 +949,16 @@ def clf(self, keep_observers=False):
 
         self.stale = True
 
-    # synonym for `clf`."""
-    clear = clf
+    # synonym for `clear`.  We do this with a property to make sure that we get
+    # a literal alias that correctly behaves for sub-classes (without resorting
+    # to __subclass_init__ or other meta-programming tools.
+    @property
+    def clf(self):
```

Comment:
Juts checking if the docstring gets copied here too?

Reply:
It does not :(

Reply:
well, `help(fig.clf)` works, but `? fig.clf` is too smart and gives the property docstring.

Reply:
If we care about the docstring, this should be good enough.
```
def  clf(self):
    """Clear the figure. Synonym for ``clear()``."""
```

Reply:
Let's not overengineer this.  Plus, this still doesn't prevent someone from accidentally overriding clf, and also breaks using unbound methods (`Figure.clear(fig)` -- probably exceptionally rare, but there's no good reason to break it...).